tests/kernel/mp/src/main.c

```diff
@@ -93,14 +98,28 @@ FUNC_NORETURN void cpu1_fn(void *arg)
  */
 ZTEST(multiprocessing, test_mp_start)
 {
-	cpu_arg = 12345;
+	for (int i = 1; i < CONFIG_MP_MAX_NUM_CPUS; i++) {
+		int wait_count;
 
-	arch_start_cpu(1, cpu1_stack, CPU1_STACK_SIZE, cpu1_fn, &cpu_arg);
+		TC_PRINT("Starting CPU #%d...\n", i);
 
-	while (!cpu_running) {
-	}
+		cpu_arg = 12345 * i;
+
+		arch_start_cpu(i, cpu_stacks[i], CPU_STACK_SIZE, cpu_fn, &cpu_arg);
 
-	zassert_true(cpu_running, "cpu1 didn't start");
+		/* Wait for about 5 (500 * 10ms) seconds for CPU to start. */
+		wait_count = 500;
+		while (!cpu_running[i]) {
+			k_busy_wait(10 * USEC_PER_MSEC);
```

Comment:
Is it safe to use timer (whis is internaly used in k_busy_wait) as this early stage?
At least we setup timers in case of SMP after we bring up secondaty CPUs cores.

Reply:
That should not be an issue. The test starts by `main()` (via ztest), so it runs after drivers are initialized and kernel is up and running. This test skips all the auto-start on other CPUs, and does it manually, hence the `CONFIG_SMP=n` in `prj.conf`.